tests/power_cycle/test_time.py

```diff
@@ -47,352 +34,99 @@ def setup_method(self):
-            attr_is_dict = isinstance(attr_in_tested_class, dict)
-            assert attr_is_dict
-
-            keys_in_attr = attr_in_tested_class.keys()
-            for key in keys_in_attr:
-                value = attr_in_tested_class[key]
-                value_as_list = validate_list(value)
-                valid_values_in_valid_config_json_values = [
-                    v in valid_config_json_values for v in value_as_list
-                ]
-                assert all(valid_values_in_valid_config_json_values)
-
-        for high_key in valid_highest_level_json_keys:
-            text_before_hiphen = high_key.split("-", 1)[0]
-            necessary_attr = "_" + text_before_hiphen + "_format"
-            necessary_attr_is_in_class = necessary_attr in all_class_attr
-            format_exists_for_high_key = necessary_attr_is_in_class
-            assert format_exists_for_high_key
+        return ScenarioBuilderConfig(**self.scenario_json_contents)
 
     def test_constructor(self):
-        tested_class = self.tested_class
-
-        scenario_json_path = self.scenario_json_path
-        sample = tested_class(scenario_json_path)
-        assert isinstance(sample, ScenarioBuilder)
-
-        scenario = sample.scenario
-        scenario_duration = scenario.duration
-        pulse_set = scenario.pulse_set
-        for pulse in pulse_set:
+        for pulse in self.sample.scenario.pulse_set:
             pulse_duration = pulse.duration
             phase_set = pulse.phase_set
             for phase in phase_set:
                 phase_duration = phase.duration
                 assert phase_duration == sum(phase.durations_list)
             assert pulse_duration == sum(pulse.durations_list)
-        assert scenario_duration == sum(scenario.durations_list)
+        assert self.sample.scenario.duration == sum(self.sample.scenario.durations_list)
 
     def test_validate_config(self):
-        tested_class = self.tested_class
-        tested_class_error = self.tested_class_error
-
-        scenario_json_contents = self.scenario_json_contents
-        valid_highest_level_json_keys = self.highest_level_json_keys
-        all_configs = self.run_validate_config()
-
-        possible_errors = (KeyError, TypeError, tested_class_error)
-        for config in all_configs:
-            config_is_dict = isinstance(config, dict)
-            assert config_is_dict
-
-        for valid_key in valid_highest_level_json_keys:
+        for valid_key in self.highest_level_json_keys:
             wrong_contents = copy_dict_with_wrong_key(
-                scenario_json_contents,
+                self.scenario_json_contents,
                 valid_key,
             )
-            with pytest.raises(possible_errors):
-                wrong_configs = tested_class._validate_config(wrong_contents)
+            with pytest.raises((KeyError, TypeError)):
+                ScenarioBuilderConfig(**wrong_contents)
 
     @pytest.mark.parametrize(
```

Comment:
This test needs to be updated to comply with the new `ModuleType` class.

Reply:
Its already covered AFAIK. the NotImplemented is just because its not implemented in the code atm